tensorflow/core/kernels/mkl_fused_batch_norm_op.cc

```diff
@@ -146,83 +168,104 @@ class MklFusedBatchNormFwdPrimitive : public MklPrimitive {
-    context_.src_mem.reset(new memory({src_md, cpu_engine_}, DummyData));
-    context_.dst_mem.reset(new memory(fwd_pd.dst_primitive_desc(), DummyData));
-
-    if (context_.flags & use_scale_shift) {
-      auto weights_desc = memory::desc({2, fwdParams.depth}, MklDnnType<U>(),
-                                       memory::format::nc);
-      context_.weights_mem.reset(
-          new memory({weights_desc, cpu_engine_}, DummyData));
+#endif  // ENABLE_MKLDNN_V1
+
+    context_.fwd_pd.reset(new BatchNormFwdPd(fwd_desc, cpu_engine_));
+
+    // Create memory primitive based on dummy data
+    context_.src_mem.reset(new MEMORY_CONSTRUCTOR(
+        context_.fwd_pd->PRIMITIVE_DESC_SRC, cpu_engine_, DummyData));
+    context_.dst_mem.reset(new MEMORY_CONSTRUCTOR(
+        context_.fwd_pd->PRIMITIVE_DESC_DST, cpu_engine_, DummyData));
+
+    memory::dims s_dims = {2, fwdParams.depth};
+    memory::dims m_dims = {1, fwdParams.depth};
+    if (IS_SET(use_scale_shift)) {
+      context_.weights_mem.reset(new MEMORY_CONSTRUCTOR_USING_MEM_PD(
+          s_dims, U, MEMORY_FORMAT::nc, cpu_engine_, DummyData));
     }
 
-    if (fwdParams.training || (context_.flags & use_global_stats)) {
-      auto mean_desc = memory::desc({1, fwdParams.depth}, MklDnnType<U>(),
-                                    memory::format::nc);
-      context_.mean_mem.reset(new memory({mean_desc, cpu_engine_}, DummyData));
+    if (fwdParams.training || (IS_SET(use_global_stats))) {
+      context_.mean_mem.reset(new MEMORY_CONSTRUCTOR_USING_MEM_PD(
+          m_dims, U, MEMORY_FORMAT::nc, cpu_engine_, DummyData));
 
-      auto variance_desc =
-          memory::desc({1, fwdParams.depth}, MklDnnType<U>(), memory::nc);
-      context_.variance_mem.reset(
-          new memory({variance_desc, cpu_engine_}, DummyData));
+      context_.variance_mem.reset(new MEMORY_CONSTRUCTOR_USING_MEM_PD(
+          m_dims, U, MEMORY_FORMAT::nc, cpu_engine_, DummyData));
     }
 
-    // BatchNorm forward primitive
-    if (!fwdParams.training && !(context_.flags & use_global_stats)) {
-      if ((context_.flags & use_scale_shift) && mkldnn_use_scaleshift) {
+    // BatchNorm forward primitive.
+#ifdef ENABLE_MKLDNN_V1
+    std::unordered_map<int, memory> net_args_map = {
+        {MKLDNN_ARG_SRC, *context_.src_mem},
+        { MKLDNN_ARG_DST,
+          *context_.dst_mem }};
+    if (fwdParams.training || IS_SET(use_global_stats)) {
+      net_args_map.insert({{MKLDNN_ARG_MEAN, *context_.mean_mem},
+                           { MKLDNN_ARG_VARIANCE,
+                             *context_.variance_mem }});
+    }
+    if (IS_SET(use_scale_shift)) {
+      net_args_map.insert({{ MKLDNN_ARG_WEIGHTS, *context_.weights_mem }});
+    }
+    context_.net_args.emplace_back(net_args_map);
+    context_.bn_fwd.reset(new batch_normalization_forward(*context_.fwd_pd));
+#else
+    if (!fwdParams.training && !(IS_SET(use_global_stats))) {
+      if ((IS_SET(use_scale_shift)) && GET_FLAG(use_scale_shift)) {
```

Comment:
What is the meaning of `&& GET_FLAG(use_scale_shift)` though? `&` makes more sense to me than `&&`, but both cases still make adding `GET_FLAG(use_scale_shift)` at the end unnecessary.
Case `&&`: `(content & flag) && flag` = `content & flag && true` (because `flag` is never 0) = `content & flag` .
Case `&`:  `(content & flag) & flag` = `content & flag`.
```suggestion
      if (IS_SET(use_scale_shift)) {
```
Same for lines 242 and 255.

Reply:
The team advised me to keep the original code as we are debugging some issues. I will rewrite/cleanup the code in a later PR.

Reply:
I understand. Please keep the original code for now. :)

Reply:
Btw, please add a todo in the code so it's easy to keep track.

Reply:
Added